File: backend/internal/model/task.go
package model

import "time"

type Task struct {
	Base
	UserID           uint       `gorm:"index;not null" json:"user_id"`
	User             User       `gorm:"foreignKey:UserID;constraint:OnDelete:CASCADE" json:"-"`
	CategoryID       *uint      `json:"category_id"`
	Category         *Category  `gorm:"foreignKey:CategoryID;constraint:OnDelete:SET NULL" json:"category"`
	Title            string     `gorm:"not null" json:"title"`
	Description      *string    `json:"description"`
	Priority         *int       `gorm:"default:2" json:"priority"`
	IsCompleted      bool       `gorm:"not null;default:false" json:"is_completed"`
	EstimatedHours   *float64   `json:"estimated_hours"`
	AiEstimatedHours *float64   `json:"ai_estimated_hours"`
	StartTime        *time.Time `gorm:"index" json:"start_time"`
	EndTime          *time.Time `json:"end_time"`
	DueDate          *time.Time `json:"due_date"`
	AchievementRate  *int       `json:"achievement_rate"`
}

File: ai/app/prompts/task.py
SYSTEM_PROMPT = (
    "あなたはプロフェッショナルなタスク見積もりアシスタントです。\n"
    "提供されたタスク情報とユーザーのコンテキストに基づき、作業にかかる見積もり時間（時間単位）を算出してください。\n"
    "出力は必ずJSON形式で、以下のキーを含めてください：\n"
    '{"estimated_hours": 1.5, "reasoning": "なぜなら..."}'
)

def build_task_estimate_prompt(title: str, description: str | None, category: str | None, user_context: str | None) -> str:
    prompt = f"タスク名: {title}\n"
    if description:
        prompt += f"詳細: {description}\n"
    if category:
        prompt += f"カテゴリ: {category}\n"
    if user_context:
        prompt += f"ユーザー特性・コンテキスト: {user_context}\n"
    return prompt